src/chat/Database.cpp

```diff
@@ -0,0 +1,38 @@
+// allkeys.cpp
+// chris busch
+
+#include "Database.hpp"
+
+namespace chat {
+
+Record *Database::add_record() {
+    Record *record;
+    if ((record = new Record) == nullptr) {
+        return nullptr;
+    }
+    if (top_ != nullptr)
+        top_->next_ = record;
+    current_ = top_ = record;
+    top_->next_ = nullptr;
+    if (record_count_ == 0)
+        first_ = top_;
+    record_count_++;
+    return current_;
+}
+
+Record *Database::reset() { return (current_ = first_); }
+
+Record *Database::next_record() {
+    if (current_ != nullptr) {
```

Comment:
Stylistically it's very common in C and C++ to not compare with `nullptr` or similar. Sorry if this is naggy, and/or probably just copied from the eliza OG code.

Reply:
Yes it's OG, will clean.